CsharpExtras/Map/Sparse/TwoDimensional/ISparseArray2D.cs

```diff
@@ -33,6 +44,8 @@ public interface ISparseArray2D<TVal>
         /// <param name="endCol">Column coordinate of the bottom-righ of the rectangle</param>
         /// <returns>A 2D array of values defind by the rectangle at the given coordinates</returns>
         TVal[,] GetArea(int startRow, int startCol, int endRow, int endCol);
+        void InsertColumns(int insertionIndex, int shiftVector);
```

Comment:
I assume the `shiftVector` in this case is basically the number of rows/columns to insert. Is that correct?
If so, I would rename the method param to be clearer - something like `columnsToAdd`.

Reply:
Sort of. I did consider that point, but I chose to continue with the terminology ``shiftVector`` because it encapsulates both the ``number`` of rows/columns to insert (i.e. the scalar component) and the direction (i.e. the plus or minus sign). Combining a scalar with a direction is succinctly described as a vector.

If you were to do it the Excel way - you'd ask for a non-negative (or even positive) number of rows to shift, and then separately a shift direction. But I don't see the merit in doing this - it adds an extra argument and extra clutter, while the vector approach is more elegant and succinct and the meaning of a vector is pretty standard.

Reply:
Ok, fair enough. In that case then I would just document what the parameter means and how to use it (i.e. how it behaves with a negative vs positive number)

Reply:
Ah! I thought I had already done that, my bad. Good catch.